2260 change filter projects form

entantoencuanto opened the pull request:
Closes #2260


## :v: What does this PR do?
* Hides submit filter projects button and autosubmits filter on changes.
* Displays a "clear filters" button redirecting to index without filter parameters.
* Fixes some links disabling turbolinks to load datepicker after redirections.
* Adds admin option to `with` context helper method for tests and updates tests

Some problems pending:

* After filter submit button vanish, part of date selector appears out of viewport:
<img width="1128" alt="Screen Shot 2019-04-30 at 22 51 56" src="https://user-images.githubusercontent.com/446459/56992592-ac380980-6b9a-11e9-8642-56bc178f402f.png">

* A bug found in other parts of the application: when a date range is selected, air-datepicker can't process the value and is rendered with errors:
<img width="278" alt="Screen Shot 2019-04-30 at 22 52 18" src="https://user-images.githubusercontent.com/446459/56992603-b2c68100-6b9a-11e9-9eb8-0c323b269325.png">

## :mag: How should this be manually tested?

Visit a plan projects tab and play with filters

## :eyes: Screenshots

### Before this PR
![2260-before](https://user-images.githubusercontent.com/446459/56992677-e43f4c80-6b9a-11e9-9653-8ce6431b86fe.gif)

### After this PR
![2260-after](https://user-images.githubusercontent.com/446459/56992696-ed301e00-6b9a-11e9-9efe-24ef3aee09c4.gif)

## :shipit: Does this PR changes any configuration file?

No

(Changes in these files might need to update the role in Ansible)

## :book: Does this PR require updating the documentation?

No

entantoencuanto committed:
Auto submit projects filter form and add link to clear filters

entantoencuanto committed:
Disable turbolinks in filter row links

entantoencuanto committed:
Update with context method to allow sign_in admins

entantoencuanto committed:
Update projects filters tests

ferblape commented:
Maybe the NaN are because instead of providing a date you are providing a date range?

amiedes commented on test/support/contexts.rb:
```diff
@@ -20,23 +20,32 @@ def with_hidden_elements
     yield(params)
   end
 
+  sign_out_admin if admin
```
I think we have to move

```ruby
sign_out_admin if admin
Capybara.reset_session! if params[:js]
```

to the `ensure` block to avoid keeping the admin logged in for the following tests if the current tests fails

entantoencuanto replied:
Done!

entantoencuanto committed:
Improve with_interval Node scope to avoid exceptions parsing argument

entantoencuanto committed:
Allow independent filters for start and end date

entantoencuanto committed:
Avoid data selection if datepicker expects a range

entantoencuanto committed:
Add allowBlank option to datepicker

entantoencuanto committed:
Allow blank in projects datepickers

entantoencuanto committed:
Ensure sign_out_admin in with context method

entantoencuanto commented:
> Maybe the NaN are because instead of providing a date you are providing a date range?

I've added some changes to `addDatePickerBehaviors`. The error is produced for date ranges and forms with more than one datepicker. I've fixed behavior for ranges and added an option to allow empty date fields.

ferblape commented:
Why the author filter is not working in this plan? http://esplugues.gobify.net/admin/plans/14/projects Projects don't have an author yet?

Commit:
Merge branch 'master' into 2260-change_filter_projects_form

entantoencuanto commented:
> Why the author filter is not working in this plan? http://esplugues.gobify.net/admin/plans/14/projects Projects don't have an author yet?

The authorship of projects imported from CSV is blank, the author of a project is assigned when an admin with edit permissions creates a project.

entantoencuanto committed:
Fix wrong translations

ferblape commented on app/models/gobierto_plans/node.rb:
```diff
@@ -14,7 +14,21 @@ class Node < ApplicationRecord
+
+      dates = [nil, nil].zip(dates).map { |date| date.compact.first }
+
+      where("starts_at >= ? AND ends_at <= ?", *dates)
```
I think I'd change this AND by an OR. This way, when I search projects between two dates I'll get all the projects active in those dates. If you use an AND you'll get all the projects that start or end inside that range, which I think is more difficult to understand from the users side.

Commit:
Permisive date range
